kobuki_ws/src/kobuki_rviz/launch/open_rviz_launch.py

```diff
@@ -0,0 +1,36 @@
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
+from launch.substitutions import LaunchConfiguration
+from launch_ros.actions import Node
+
+ARGUMENTS = [
+    DeclareLaunchArgument(
+        name="use_sim_time",
+        default_value="True",
+        description="Use simulation time",
+    ),
+    DeclareLaunchArgument(
+        name="rviz_config_path",
+        default_value="",
+        description="The path of .rviz file",
+    ),
+]
+
+
+# This launch file will only launch rviz2 node.
+# It is used when other repos want to launch rviz2 for kobuki. For instance: kobuki_navigation
+def generate_launch_description():
+    # Launch rviz2.
+    node_rviz = Node(
+        package="rviz2",
+        executable="rviz2",
+        name="rviz",
+        output="screen",
+        arguments=["-d", LaunchConfiguration("rviz_config_path")],
+        parameters=[{"use_sim_time": LaunchConfiguration("use_sim_time")}],
+    )
+
+    ld = LaunchDescription(ARGUMENTS)
+    ld.add_action(node_rviz)
+
+    return ld
```

Comment:
The coding convention I followed comes from the [launch file of Husky](https://github.com/husky/husky/blob/729f8aa45ccd86fa33a05e07ef698c52c451cd9c/husky_gazebo/launch/gazebo.launch.py). Compared to the coding convention used in [TurtleBot3](https://github.com/ROBOTIS-GIT/turtlebot3/blob/a7dd05ae176f3f3778b0a36f7065dc9655b050e3/turtlebot3_navigation2/launch/navigation2.launch.py) or the [ROS2 tutorial](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Using-ROS2-Launch-For-Large-Projects.html), I prefer to use the current one. It is easier to scale because it is simpler to debug.

For example, the use of the pre-defined `ARGUMENTS` can help me quickly identify which parameters can be adjusted, and the use of `ld = LaunchDescription` allows me to easily enable or disable a specific node by modified only one line of code.

Reply:
However, I agree that using the coding convention from the official tutorial may make the code more readable and simpler sometimes, especially if there are a large number of files. 

Both coding conventions have their reasons. I'm not sure which one I should use here.
What are your thoughts on this?

Reply:
Thanks for the references. I also prefer the current style. To summarize:

- The `ARGUMENTS` allow quickly view the adjustable parameters.
  - In contrast to [ROS2 tutorial](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Using-ROS2-Launch-For-Large-Projects.html) where we need to scroll down to the `return` part.
  - In contrast to [Nav 2](https://github.com/ros-planning/navigation2/blob/e2c8555ad352d69aa3da1e670b888d00f6eaeb3a/nav2_bringup/launch/navigation_launch.py) where each parameters are stored in a variable and added later. I don't think the naming of these parameters are necessary. Also, adding them later is error prone and some parameters may be easily overlooked.
- The `ld = LaunchDescription` allow easier debugging by adding/removing a node by modifying a single line (and without require scrolling to find the node). It also allows quickly view all nodes used in the launch file without needed to scroll a lot.
  - This style is slightly more error prone than the style in [ROS2 tutorial](https://docs.ros.org/en/humble/Tutorials/Intermediate/Launch/Using-ROS2-Launch-For-Large-Projects.html), since we need to name each node and remember to add them afterwards.
  - However, I think the benefits outweigh the drawbacks.